src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -407,10 +422,18 @@ private void GenerateMetadataNodes()
                         // we just pull in methods that have attributes on them.
                         if ((method.Attributes & MethodAttributes.MemberAccessMask) == MethodAttributes.Public &&
                             (method.Attributes & MethodAttributes.Static) != 0 &&
+                            method.GetParameters().Count > 0 &&
                             method.GetCustomAttributes().Count > 0)
                         {
-                            var definition = new MetadataDefinition(
-                                MetadataDefinitionKind.Member, _metadataReader.GetString(method.Name));
+                            _containsExtensionsMethod = true;
+
+                            // Decode method signature to get the target type name (i.e. type name for the first parameter)
+                            var blob = _metadataReader.GetBlobReader(method.Signature);
+                            var decoder = new SignatureDecoder<ParameterTypeInfo, object>(ParameterTypeInfoProvider.Instance, _metadataReader, genericContext: null);
+                            var signature = decoder.DecodeMethodSignature(ref blob);
+                            var firstParameterTypeInfo = signature.ParameterTypes[0];
```

Comment:
can you ask tomas where we should file requests/bugs?   it's really unfortunate we can't decode this without allocating a full array here for every method we examine.

Reply:
@tmat ?

Reply:
You can indeed decode the first type without allocating.

The implementation of [`DecodeMethodSignature`](https://github.com/dotnet/corefx/blob/d3911035f2ba3eb5c44310342cc1d654e42aa316/src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/SignatureDecoder.cs#L165) doesn't do anything magical. You can "inline" it and stop decoding after the first type.

Reply:
Ew, but ok.  If you do this, make sure to link back to this code.

Reply:
this should be done.

Reply:
See this comment: https://github.com/dotnet/roslyn/pull/39126#discussion_r339848845